swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -561,6 +569,20 @@ predicate readStep(Node node1, ContentSet c, Node node2) {
     node2.asExpr() = tuple and
     c.isSingleton(any(Content::TupleContent tc | tc.getIndex() = tuple.getIndex()))
   )
+  or
+  // read of an enum member via `case let .variant(v1, v2)` pattern matching
+  exists(Expr enumExpr, ParamDecl enumField, VarDecl boundVar |
+    node1.asExpr() = enumExpr and
+    node2.asDefinition().getSourceVariable() = boundVar and
+    c.isSingleton(any(Content::EnumContent ec | ec.getField() = enumField))
+  |
+    exists(EnumElementPattern enumPat, NamedPattern namePat, int idx |
+      enumPat.getMatchingExpr() = enumExpr and
+      enumPat.getElement().getParam(idx) = enumField and
+      namePat.getIdentityPreservingEnclosingPattern*() = enumPat.getSubPattern(idx) and
+      namePat.getVarDecl() = boundVar
+    )
```

Comment:
So the role of this nested `exists` is to map the `enumExpr` to the `boundVar`, right? And we need to ensure that we're not grabbing it out of a bunch of nested patterns that should all have have `readStep`s, correct?

Is this complexity due to the fact that we don't have `DataFlow::Node`s for these nested patterns? Would it be better to add those dataflow nodes instead?

Reply:
Yeah, I think you're right. This may  be easier with Dataflow nodes for the intermediate pattern matches (already CfgNodes I think).

Reply:
Agreed. I'm not sure if the set of `DataFlow::Nodes` we currently have suffices for this. You might need to define a new branch in https://github.com/github/codeql/blob/main/swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPrivate.qll#L63 that wraps the `CfgNodes` corresponding to `Pattern`s. I don't think we'd need any local flow through these, but they really only need to exist to facilitate `readStep`s.